src/OpenSage.Game/Audio/AudioBuffer.cs

```diff
@@ -22,43 +22,48 @@ public AudioBuffer()
         public AudioBuffer(WavFile wavFile) : this()
         {
             int format = 0;
-
-            switch (wavFile.Channels)
+            if ((AudioFormatType) wavFile.AudioFormat != AudioFormatType.Microsoft)
```

Comment:
Unfortunately BitsPerSample represents the compressed bits per sample, not the uncompressed bits per sample, so for ADPCM it is 4. But you're right, it's probably not the right place to put this code, so I have opted to make BitsPerSample a method that returns the 'true' uncompressed bits per sample rather than polluting AudioBuffer.